features/steps/test_starter/{{ cookiecutter.repo_name }}/pyproject.toml

```diff
@@ -1,7 +1,41 @@
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "{{ cookiecutter.python_package }}"
+readme = "README.md"
+dynamic = ["dependencies", "version"]
+
+[project.scripts]
+{{ cookiecutter.repo_name }} = "{{ cookiecutter.python_package }}.__main__:main"
+
+[project.optional-dependencies]
+docs = [
+    "docutils<0.18.0",
+    "sphinx~=3.4.3",
+    "sphinx_rtd_theme==0.5.1",
+    "nbsphinx==0.8.1",
+    "nbstripout~=0.4",
+    "sphinx-autodoc-typehints==1.11.1",
+    "sphinx_copybutton==0.3.1",
+    "ipykernel>=5.3, <7.0",
+    "Jinja2<3.1.0",
+    "myst-parser~=0.17.2",
+]
+
+[tool.setuptools.dynamic]
+dependencies = {file = "src/requirements.txt"}
+version = {attr = "{{ cookiecutter.python_package }}.__version__"}
+
+[tool.setuptools.packages.find]
+where = ["src"]
+namespaces = false
+
 [tool.kedro]
-project_name = "{{ cookiecutter.project_name }}"
-project_version = "{{ cookiecutter.kedro_version }}"
 package_name = "{{ cookiecutter.python_package }}"
+project_name = "{{ cookiecutter.project_name }}"
+kedro_init_version = "{{ cookiecutter.kedro_version }}"
```

Comment:
`project_version` was deprecated, unsure when the switch to `kedro_init_version` was supposed to happen... good to do it now?

Reply:
Good catch! I don't think we have a ticket for removing `project_version`, let's create it or just remove `project_version` in the next PR (might be less work than creating the ticket)

Reply:
Done #3084 took the liberty of setting it to "High", we can bring back to Inbox if needed.